### PR TITLE
fix(css/modules): Fix handling of `:global()` selector

### LINKED ISSUE #8179: CSS module :global function syntax
### Describe the bug

Currently SWC css module pass cannot handle cases like this 

```
.a:global(.b .c).d {
  color: red
}
```

SWC ignores `.d`

I tried fix this, but current algorithm seems hard to implement

[lightningcss demo](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Atrue%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22.a%3Aglobal(.b%20.c).d%20%7B%5Cn%20%20color%3A%20red%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)

### Input code

_No response_

### Config

_No response_

### Playground link (or link to the minimal reproduction)

None

### SWC Info output

_No response_

### Expected behavior

.d should be renamed

### Actual behavior

.d stay as is

### Version

1.3.95

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kdy1**
**Related issue:**

 - Closes #8179

### CODE REVIEW

File: crates/swc_css_modules/tests/fixture/issue-8179.compiled.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.__local__a.b.d .__local__a.c.d {
```
Comment (kdy1) [inline crates/swc_css_modules/tests/fixture/issue-8179.compiled.css:1]: @JSerFeng What's the expected output?

[The playground link](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Atrue%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22.a%3Aglobal(.b%20.c).d%20%7B%5Cn%20%20color%3A%20red%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D) you posted says 

```css
.EgL3uq_a.b .c.EgL3uq_d{color:red}
```

but I'm not sure if it's correct.

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_css_modules

File: crates/swc_css_modules/tests/fixture/issue-8179.compiled.css (hunk 1 above)
Comment (JSerFeng) [reply crates/swc_css_modules/tests/fixture/issue-8179.compiled.css:1]: I think lightningcss is right, postcss is the same with lightningcss

Comment (JSerFeng) [reply crates/swc_css_modules/tests/fixture/issue-8179.compiled.css:1]: image

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_css_modules

**Review by JSerFeng — APPROVED** [review]

**Comment by JSerFeng** [comment]
Thanks !

**Pull request merged**